src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -50,21 +49,22 @@ private bool ConvertExpressionToType(ExpressionSyntax expression, [NotNullWhen(t
                     return false;
             };
         }
-        private PatternSyntax ParsePattern(Precedence precedence, bool afterIs = false, bool whenIsKeyword = false)
+
+        private PatternSyntax ParsePattern(Precedence precedence, bool afterIs = false, bool whenIsKeyword = false, bool exitOnColon = false)
```

Comment:
what's the impact if we jsut always pass true for this?

Reply:
I tried. There are several tests, that specificly assert that. The result if we exit pattern parsing on colon in such cases is worse. There are also several situations, where parser is allowed to treat `:` as a separator in a pattern list for better error recovery. It seems, that this is a pretty comon typing scenario when you add another property pattern before exisiting one, e.g. `{ Prop: $$ Prop2: { ... } }`. So I limited this behaviour only to switch statements

Reply:
great.  can you doc the code as such?